pulsar-broker/src/main/java/org/apache/pulsar/broker/web/PulsarWebResource.java

```diff
@@ -176,9 +178,23 @@ protected void validateSuperUserAccess() {
             validateOriginalPrincipal(pulsar.getConfiguration().getProxyRoles(), appId, originalPrincipal);
 
             if (pulsar.getConfiguration().getProxyRoles().contains(appId)) {
-                Set<String> superUserRoles = pulsar.getConfiguration().getSuperUserRoles();
-                boolean proxyAuthorized = superUserRoles.contains(appId);
-                boolean originalPrincipalAuthorized = superUserRoles.contains(originalPrincipal);
+
+                boolean proxyAuthorized;
+                boolean originalPrincipalAuthorized;
+
+                try {
+                    proxyAuthorized = pulsar.getBrokerService()
```

Comment:
minor, but since both of these could potentially be async computations, it'd be better to kick them both off, and then wait for completion.

Reply:
will fix